internal/terraform/lang/types.go

```diff
@@ -76,8 +76,26 @@ type CompletionCandidate interface {
 	Label() string
 	Detail() string
 	Documentation() MarkupContent
-	Snippet() string
-	PlainText() string
+	Snippet() TextEdit
```

Comment:
I think this is fine for now, but probably a lot easier to just make this function `TextEdit(snippetSupport bool)` and then you can take the branch out downstream.

Reply:
Hmm, I see your point, but also I feel that the two methods communicate the situation better than one with a boolean argument. It seems fairly cheap here to enforce the [single responsibility principle](https://blog.gfader.com/2010/06/avoid-boolean-parameters-in-method.html) compared to maintaining if statements in multiple places.

In theory compiling the snippets may become more expensive/complex for some blocks and complex code deserves its own function to retain some readability. 😉 

Maybe it's premature optimization, time will tell 🤷‍♂️